app/views/SelectListView.js

```diff
@@ -0,0 +1,185 @@
+/* eslint-disable no-mixed-spaces-and-tabs */
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+	View, StyleSheet, FlatList, Text, Alert
+} from 'react-native';
+import { connect } from 'react-redux';
+import { HeaderBackButton } from '@react-navigation/stack';
+import * as List from '../containers/List';
+
+import { leaveRoom as leaveRoomAction } from '../actions/room';
+import sharedStyles from './Styles';
+import I18n from '../i18n';
+import * as HeaderButton from '../containers/HeaderButton';
+import StatusBar from '../containers/StatusBar';
+import { themes } from '../constants/colors';
+import { withTheme } from '../theme';
+import SafeAreaView from '../containers/SafeAreaView';
+import { animateNextTransition } from '../utils/layoutAnimation';
+import Loading from '../containers/Loading';
+
+const styles = StyleSheet.create({
+	buttonText: {
+		fontSize: 17,
+		margin: 16,
+		...sharedStyles.textRegular
+	}
+});
+
+class SelectListView extends React.Component {
+	static propTypes = {
+		navigation: PropTypes.object,
+		route: PropTypes.object,
+		user: PropTypes.shape({
+			id: PropTypes.string,
+			token: PropTypes.string
+		}),
+		theme: PropTypes.string,
+		isMasterDetail: PropTypes.bool
+	};
+
+	constructor(props) {
+		super(props);
+		const data = props.route?.params?.data;
+		this.title = props.route?.params?.title;
+		this.teamName = props.route?.params?.teamName;
+		this.nextAction = props.route?.params?.nextAction;
+		this.state = {
+			data,
+			selected: [],
+			loading: false
+		};
+		this.setHeader();
+	}
+
+	setHeader = () => {
+		const { navigation, isMasterDetail, theme } = this.props;
+		const { selected } = this.state;
+
+		const options = {
+			headerShown: true,
+			headerTitleAlign: 'center',
+			headerTitle: I18n.t(this.title)
+		};
+
+		if (isMasterDetail) {
+			options.headerLeft = () => <HeaderButton.CloseModal navigation={navigation} />;
+		} else {
+			options.headerLeft = () => <HeaderBackButton labelVisible={false} onPress={() => navigation.pop()} tintColor={themes[theme].headerTintColor} />;
+		}
+
+		options.headerRight = () => (
+			<HeaderButton.Container>
+				<HeaderButton.Item title={I18n.t('Next')} onPress={() => this.nextAction(selected)} testID='select-list-view-submit' />
+			</HeaderButton.Container>
+		);
+
+		navigation.setOptions(options);
+	}
+
+	renderHeader = () => {
+		const { theme } = this.props;
+		return (
+			<View style={{ backgroundColor: themes[theme].backgroundColor }}>
+				<Text style={[styles.buttonText, { color: themes[theme].bodyText }]}>{I18n.t('Select_Teams')}</Text>
```

Comment:
This is a reusable component. Why `select teams`?

Reply:
Solved